README.md

```diff
@@ -99,6 +92,65 @@ Both the executable and the kernel module (and OCTF if submodule is present) are
 sudo make install
 ~~~
 
+<a id="theory_of_operation"></a>
+
+## Theory of operation
+
+Standalone Linux IO Tracer captures request data by registering to multiple trace points surfaced
+by the Linux kernel (e.g. BIO queueing, BIO splitting, BIO completion). This allows for gathering of
+IO metadata at the request level and passing it between kernel and userspace.
+
+A circular buffer is allocated and shared between kernel and userspace for each logical CPU core and
+trace events (example shown below) are then pushed into it.
+
+```c
+struct iotrace_event_device_desc {
+    /** Event header */
+    struct iotrace_event_hdr hdr;
+
+    /** Device Id */
+    uint64_t id;
+
+    /** Device size in sectors */
+    uint64_t device_size;
+
+    /** Canonical device name */
+    char device_name[32];
+
+    /** Device model */
+    char device_model[256];
+} __attribute__((packed, aligned(8)));
+```
```

Comment:
Let's add link to trace events declaration file, for more details if user would like to see